src/attr.c

```diff
@@ -460,7 +471,7 @@ static int push_attr_file(
 static int push_one_attr(void *ref, const char *path)
 {
 	attr_walk_up_info *info = (attr_walk_up_info *)ref;
-	git_attr_file_source src[2];
+	git_attr_file_source src[GIT_ATTR_FILE_NUM_SOURCES];
```

Comment:
This is weird. We had three sources previously (in memory, from file, from index). Was this simply wrong and could've led to out-of-bounds access?

Reply:
In this codepath we call `attr_decide_sources` which will only include the source `GIT_ATTR_FILE__FROM_FILE` and/or `GIT_ATTR_FILE__FROM_INDEX`, and may also include `GIT_ATTR_FILE__FROM_HEAD`.  This will over-allocate by at least one, but that seems acceptable for the readability and maintainability trade-off.